src/apps/Permissions/PermissionsView.js

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import {
+  ButtonIcon,
+  ContextMenu,
+  ContextMenuItem,
+  DataView,
+  IconTrash,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { usePermissions } from '../../contexts/PermissionsContext'
+import LocalIdentityBadge from '../../components/IdentityBadge/LocalIdentityBadge'
+import PermissionsIdentityBadge from './PermissionsIdentityBadge'
+
+function PermissionsView({ permissions, onManageRole, heading, showApps }) {
+  const { layoutName } = useLayout()
+  const willRenderEntryChild = useMemo(
+    () => permissions.some(permission => permission.entities.length > 1),
+    [permissions]
+  )
+
+  const fields = [
+    'Action',
+    'On app',
+    { label: 'Assigned to entity', childStart: true },
+    'Managed by',
+  ]
+
+  if (!showApps) {
+    fields.splice(1, 1)
+  }
+
+  return (
+    <DataView
+      heading={heading}
+      mode={
+        layoutName === 'large' || (layoutName === 'medium' && !showApps)
+          ? 'table'
+          : 'list'
+      }
+      fields={permissions.length ? fields : []}
+      entries={permissions}
+      renderEntry={entry => renderEntry(entry, showApps)}
+      renderEntryChild={willRenderEntryChild ? renderEntryChild : undefined}
```

Comment:
This is here to avoid the extra space in case no entry children are rendered no? Should this be inner behaviour or should we make it explicit to be aware of this?

Reply:
Yes, it's to avoid the extra space.

It may be tough to detect correctly in `DataView` all the time, since it might not know exactly when children will be rendered unless we pass more props in, which is more or less the same as we're doing here. But perhaps a prop would be more explicit when we want to turn off the children behaviour 🤷‍♂; the current `DataView` doc doesn't go into great detail on when this spacing is applied.

Reply:
Maybe a check for `renderEntryChild === null` and then not add the extra column?

Reply:
It is to ensure that the extra space is consistent, and is present if any child could be displayed with the current set of data, even if not on the current page. `DataView` can’t know about it because we only pass a subset of the data, so that’s how we inform it. The children behavior of `DataView` can be turned off by setting `renderEntryChild` to `undefined` (or not setting it), which is what we are doing here.